hamcrest/Hamcrest/Type/IsNumeric.php

```diff
@@ -19,10 +19,42 @@ public function __construct()
 
     public function matches($item)
     {
+        if ($this->is_hexadecimal($item)) {
+
+            return true;
+        }
+
         return is_numeric($item);
     }
 
     /**
+     * This check is necessary because PHP 7 doesn't recognize hexadecimal string as numeric anymore.
+     *
+     * If $item is a string and that the string is long enough to be hexadecimal
+     * Check if it's hexadecimal ($item[1] === 'x')
+     * And if it is, decode it before to return it to is_numeric.
+     *
+     * @param mixed $item
+     * @return boolean
+     */
+    private function is_hexadecimal($item)
```

Comment:
Any specific reason to name method in a snake case? If library coding standard enforces that, then of course keep it, but otherwise change to camelCase.